src/main/java/org/apache/maven/plugins/shade/mojo/ShadeMojo.java

```diff
@@ -823,6 +844,50 @@ private File resolveArtifactSources( Artifact artifact )
             }
         }
 
+        // first check for backward compatibility this is not already configured explicitly
+        boolean addExclusion = true;
```

Comment:
We need a similar trick here:

    public Collection<Filter> getFilters()
    {
      if ( this.filters == null )
      {
        return defaultFilters();
      }
      else
      {
        return this.filters;
      }
    }

Reply:
likely the same